test/e2e/basic/rollback.go

```diff
@@ -0,0 +1,96 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+// Test the functionality of the preStop container lifecycle hook -- Tuned settings rollback.
+var _ = ginkgo.Describe("[basic][rollback] Node Tuning Operator settings rollback", func() {
+	const (
+		profileIngress  = "../../../examples/ingress.yaml"
+		podLabelIngress = "tuned.openshift.io/ingress"
+		sysctlVar       = "net.ipv4.tcp_tw_reuse"
+	)
+
+	ginkgo.Context("Tuned settings rollback", func() {
+		var (
+			pod *coreapi.Pod
+		)
+
+		// Cleanup code to roll back cluster changes done by this test even if it fails in the middle of ginkgo.It()
+		ginkgo.AfterEach(func() {
+			ginkgo.By("cluster changes rollback")
+			if pod != nil {
+				util.ExecAndLogCommand("oc", "label", "pod", "--overwrite", "-n", ntoconfig.OperatorNamespace(), pod.Name, podLabelIngress+"-")
+			}
+			util.ExecAndLogCommand("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "-f", profileIngress)
```

Comment:
That is correct.  Both the Pod unlabelling and the deletion will be performed twice on success, but it is intended to keep the test code as simple as possible.  I keep the `ginkgo.AfterEach()` in all tests to "rollback" the changes at any point of the tests (mostly failures).  I guess your point it so change the code not to perform the rollback on success, right?  But then we come to the simplicity argument.  I'd rather keep this as is, personally.

Reply:
I take back what I said. :)  This needs to be written better.  The Tuned pod will be deleted again completely unnecessarity when the tests succeeds.  Will fix this.  Thanks for bringing it up.

Reply:
the way I would personally code it is to keep the rollback code, but rely on it in the main test block, something like this:

```python
def cleanup(): 
  if pod:
      // deleting Pod
  // deleting custom profile


// creating custom profile
// ensuring the custom worker node profile was set
// deleting Pod
pod = nil
// waiting for a new tuned Pod to be ready on node
// ensuring the original %s value (%s) is set in Pod %s

// implicit call to cleanup()
```
so that you don't code the cleanup twice

...... but that's bikeshedding, only personal tastes if the clean-up calls are silent :)

Reply:
I do not disagree with the `pod = nil` once you deleted the object.  That really should be there and will fix. :)